### PR TITLE
Add implementation of S3Cache

### PR DESCRIPTION
**Pull request by yolken**
to: @mistercrunch 

#### Content

This change implements an `S3Cache` which extends from `werkzeug.contrib.cache.BaseCache`. All S3 calls are made using the `boto3` library.

### CODE REVIEW

File: superset/results_backends.py [hunk 1]
```diff
@@ -7,28 +7,123 @@
...
+            value_file = StringIO.StringIO()
+
+            try:
+                self.s3_client.download_fileobj(
+                    self.bucket,
+                    self._full_s3_key(key),
+                    value_file
+                )
+            except Exception as e:
+                logging.warn('Exception while trying to get %s: %s', key, e)
```
Comment (mistercrunch) [inline superset/results_backends.py]: let's `logging.exception(e)` to get the full stack trace

File: superset/results_backends.py [hunk 2]
```diff
@@ -7,28 +7,123 @@
...
+                    Delete={
+                        'Objects': [
+                            {
+                                'Key': self._full_s3_key(key)
+                            }
+                        ]
+                    }
+                )
+            except Exception as e:
+                logging.warn('Exception while trying to delete %s: %s', key, e)
```
Comment (mistercrunch) [inline superset/results_backends.py]: same as above

File: superset/assets/version_info.json [hunk 3]
```diff
@@ -1 +1 @@
-{"GIT_SHA": "2d08e240285288b71df98747ddd4b6cca3220c5a", "version": "0.15.2"}
\ No newline at end of file
+{"GIT_SHA": "1e94498d9d548cbea6466a45dafa3b919c65bd1f", "version": "0.15.4"}
```
Comment (mistercrunch) [inline superset/assets/version_info.json]: we should `git rm --cached` this file and add it to the .gitignore list... (my bad)

**Review by mistercrunch — APPROVED** [review]
A few very minor nits, but otherwise looks amazing. LGTM

File: superset/results_backends.py (hunk 1 above)
Comment (yolken) [inline superset/results_backends.py]: Fixed.

File: superset/results_backends.py (hunk 2 above)
Comment (yolken) [inline superset/results_backends.py]: Fixed.

File: superset/assets/version_info.json (hunk 3 above)
Comment (yolken) [inline superset/assets/version_info.json]: Done.

**Comment by mistercrunch** [comment]
LGTM

**Comment by santhavathi** [comment]
@yolken, do we have the option of saving result set in superset to S3.

**Pull request merged**